src/sentry/integrations/jira/models/create_issue_metadata.py

```diff
@@ -156,8 +157,8 @@ def from_dict(cls, data: dict[str, Any]) -> JiraIssueTypeMetadata:
         )
 
     @classmethod
-    def from_jira_meta_config(cls, meta_config: dict[str, Any]) -> list[JiraIssueTypeMetadata]:
+    def from_jira_meta_config(cls, meta_config: dict[str, Any]) -> dict[str, JiraIssueTypeMetadata]:
         issue_types_list = meta_config.get("issuetypes", {})
         issue_configs = [cls.from_dict(it) for it in issue_types_list]
 
-        return issue_configs
+        return {it.id: it for it in issue_configs}
```

Comment:
This makes parsing a single large config and finding an issue type a bit easier.